src/backend.py

```diff
@@ -918,6 +918,44 @@ def create_account(self, authorisor: Member, owner_id: int, economy: Economy, na
         self.session.commit()
         return account
 
+    def transfer_ownership(self, authorisor, account: Account, new_owner_id: int):
+        '''
+        Transfers the ownership of an account from one user to another.
+        :param authorisor: The initiator of this action.
+        :param account: The account whose ownership will be changed.
+        :param new_owner_id: The new owner's user ID.
+        :returns: The account with the new owner.
+        '''
+
+        # I'm gonna treat transferring accounts like closing them because technically the user is
+        # closing that account and giving it to somebody else
+        if not self.has_permission(authorisor, Permissions.CLOSE_ACCOUNT, account=account):
+            raise BackendError("You do not have permission to transfer the ownership of this account.")
+
+        if account.account_type == AccountType.USER:
+            raise BackendError("You cannot transfer your user account.")
+
+        old_owner_id = account.owner_id
+        economy = account.economy
+        try:
+            account.owner_id = new_owner_id
+
+            self.session.add(Transaction(
+                actor_id = authorisor.id,
+                economy_id = economy.economy_id if economy is not None else None,
+                action = Actions.UPDATE_ACCOUNTS,
+                cud = CUD.UPDATE,
+                meta = make_serializable({
+                    "old_account_owner": old_owner_id,
+                    "new_account_owner": new_owner_id
+                })
+            ))
+        except Exception as e:
+            self.session.rollback()
+            raise BackendError(f"Could not transfer account's ownership from <@!{old_owner_id}> to <@!{new_owner_id}: {e}.")

```

Comment:
I'd leave out the mentions in this string, sometimes accounts are owned by roles and need to be @'ed using <@&{id}> instead, there is currently no way to distinguish between role and user ids without hitting the discord apis so a more generic message like "Failed to transfer ownership" is probably better